pkg/ccl/importccl/exportcsv_test.go

```diff
@@ -482,3 +484,44 @@ func TestExportPrivileges(t *testing.T) {
 	_, err = testuser.Exec(`EXPORT INTO CSV $1 FROM TABLE privs`, dest)
 	require.NoError(t, err)
 }
+
+func TestExportTargetFileSizeSetting(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	ctx := context.Background()
+	dir, cleanup := testutils.TempDir(t)
+	defer cleanup()
+
+	tc := testcluster.StartTestCluster(
+		t, 1, base.TestClusterArgs{ServerArgs: base.TestServerArgs{ExternalIODir: dir}})
+	defer tc.Stopper().Stop(ctx)
+
+	// Every row is 3 bytes, and so the table will be ~25KB. After CSV encoding
+	// every row is 6 bytes, so the total number of CSV bytes is ~50KB.
+	const rows = 8533
+	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method == "GET" {
+			for i := 0; i < rows; i++ {
+				fmt.Fprintln(w, 'a')
+			}
+		}
+	}))
+	defer srv.Close()
+
+	conn := tc.Conns[0]
+	sqlDB := sqlutils.MakeSQLRunner(conn)
+
+	sqlDB.Exec(t, `CREATE TABLE foo (x BYTES);`)
+	sqlDB.Exec(t, `IMPORT INTO foo (x) CSV DATA ($1)`, srv.URL)
+
+	sqlDB.Exec(t, `EXPORT INTO CSV 'nodelocal://0/foo' WITH chunk_size='10KB' FROM TABLE foo`)
+	files, err := ioutil.ReadDir(filepath.Join(dir, "foo"))
+	require.NoError(t, err)
+	require.Equal(t, 5, len(files))
+
+	sqlDB.Exec(t, `EXPORT INTO CSV 'nodelocal://0/foo-compressed' WITH chunk_size='10KB',compression='gzip' FROM TABLE foo`)
+	zipFiles, err := ioutil.ReadDir(filepath.Join(dir, "foo-compressed"))
+	require.NoError(t, err)
+	require.Equal(t, 1, len(zipFiles))
```

Comment:
hm, we should pick a chunk size s.t. we get >1 file to prove it works. hopefully I didn't mislead you too much with shrinking the initial dataset if it is now <36kb compressed.

Reply:
Sorry, I meant to add a comment but got pulled into something else. We seem to be compressing really well and always staying at 10 bytes even at our original dataset and slightly bigger. I'm trying to find a sweetspot at which we get >1 file.

Reply:
might need to write some bytes other than `a`

Reply:
you might be able to just `EXPORT ... generate_series()` ?

Reply:
'a' was compressing too well so I switched to monotonically increasing int. I observed that gzip doesn't necessarily buffer up until 32Kib. The first of the two files was ~20KiB when it was flushed to the bytes buffer.

Reply:
Changed the test to use generate series so that we do not start an IMPORT job. The gzip flushing results in 6 or 7 files under race, so I changed the test condition to check for greater than 6 files.